src/markdown-pages/collect-data/opentelemetry-manual/instrument.mdx

```diff
@@ -0,0 +1,641 @@
+
+Here, you manually instrument your application to generate traces with the OpenTelemetry [API](https://github.com/open-telemetry/opentelemetry-python/tree/main/opentelemetry-api). Then, you configure the [SDK](https://github.com/open-telemetry/opentelemetry-python/tree/main/opentelemetry-sdk) to send them to New Relic so you can analyze the results later.
+
+## Instrument your application
+
+<Steps>
+
+<Step>
+
+In the terminal window that's running your simulator, press `<CTRL-C>`.
+
+You should see your simulator shut down. Now you can add some dependencies and update your app logic.
+
+</Step>
+
+<Step>
+
+Install the OpenTelemetry SDK and supporting packages into your virtual environment:
+
+<>
+
+```bash
+pip install opentelemetry-api
+pip install opentelemetry-sdk
+pip install opentelemetry-exporter-otlp-proto-grpc
+```
+
+</>
+
+</Step>
+
+<Step>
+
+In _db.py_, create a [tracer provider](https://opentelemetry.io/docs/reference/specification/trace/api/#tracerprovider):
+
+<>
+
+```py fileName=db.py
+import logging
+from opentelemetry import trace
+from opentelemetry.sdk.resources import Resource
+from opentelemetry.sdk.trace import TracerProvider
+
+provider = TracerProvider(
+    resource=Resource.create({"service.name": "speedeedeebee"})
+)
+trace.set_tracer_provider(provider)
+
+class DuplicateKeyError(Exception):
+    pass
+
+class KeyDoesNotExistError(Exception):
+    pass
+
+db = {}
+
+def read(key):
+    """Read key from the database."""
+    global db
+
+    try:
+        value = db[key]
+        logging.debug("Successful read")
+        return value
+    except KeyError as ke:
+        msg = f"Key `{key}` doesn't exist"
+        logging.debug(msg)
+        raise KeyDoesNotExistError(msg)
+
+def create(key, value):
+    """Write key:value to the database."""
+    global db
+
+    if key in db:
+        msg = f"Key `{key}` already exists"
+        logging.debug(msg)
+        raise DuplicateKeyError(msg)
+
+    db[key] = value
+    logging.debug("Successful create")
+    return value
+
+def update(key, value):
+    """Update key in the database."""
+    global db
+
+    if key in db:
+        db[key] = value
+        logging.debug("Successful update")
+        return value
+
+    msg = f"Key `{key}` doesn't exist"
+    logging.debug(msg)
+    raise KeyDoesNotExistError(msg)
+
+def delete(key):
+    """Delete key from the database."""
+    global db
+
+    if key in db:
+        del db[key]
+        logging.debug("Successful delete")
+        return True
+
+    return False
+```
+
+</>
+
+<br />
+
+A tracer provider is used for holding configurations and for building [tracers](https://opentelemetry.io/docs/reference/specification/trace/api/#tracer). Tracers are then used for creating [spans](https://opentelemetry.io/docs/reference/specification/trace/api/#span). Spans collect information about an operation or process.
```

Comment:
I think it'll be more useful to move the introduction to trace provider before we create one in the application.